### PR TITLE
Filter system host when is isRequired

### LINKED ISSUE #28924: if a host/folder field is marked as required, then we do not allow/show SYSTEM_HOST
Parent Issue => https://github.com/dotCMS/core/issues/28494

### Screenhosts

if a host/folder field is marked as required, then we do not allow/show SYSTEM_HOST

Image

if a host/folder field is marked as not required, then we do allow/show SYSTEM_HOST

Image

### PR DESCRIPTION
**Pull request by nicobytes**
If a host/folder field is marked as required, then we do not allow/show SYSTEM_HOST

### Proposed Changes
* Add isRequired attr to enable filter

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

This PR fixes: #28924

### CODE REVIEW

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/methods/loadSites.ts [hunk 1]
```diff
@@ -11,14 +11,33 @@ import type { HostFolderFiledState } from './../host-folder-field.store';
...
+                            if (isRequired) {
+                                const index = sites.findIndex(
+                                    (item) => item.label === SYSTEM_HOST_NAME
+                                );
+                                if (index !== -1) {
+                                    sites.splice(index, 1);
+                                }
+                            }
+
+                            return sites;
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/methods/loadSites.ts:39]: Maybe this can be one filter

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/methods/loadSites.ts (hunk 1 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/methods/loadSites.ts]: @zJaaal Great suggestion, already done!

**Review by zJaaal — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/methods/loadSites.spec.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            isRequired: false
+        };
+        rxMethod(props);
+        const hasSystemHost = store.tree().some((item) => item.label === SYSTEM_HOST_NAME);
+        expect(hasSystemHost).toBe(true);
+    });
+
+    it('should not include System Host when isRequired is true.', () => {
+        service.getSitesTreePath.mockReturnValue(of(TREE_SELECT_SITES_MOCK));
+        const rxMethod = TestBed.runInInjectionContext(() => loadSites(store, service));
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/methods/loadSites.spec.ts:38]: I was wondering how this was done.

**Review by oidacra — APPROVED** [review]

**Comment by fmontes** [comment]
Are we sure this is the right approach? Because why required will exclude the system host?

What load without a value and if is marked as required user will have to add a value and that value could be system host.

If we don't want the user to select system host we can do something else? Blacklist maybe? 

I don't know all the details behind this reasoning, but this is a component we might be using in a lot of places.

**Comment by dsilvam** [comment]
@fmontes RE: don't show system host if required: 
tagged you on Slack in the thread were this was discussed.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/methods/loadSites.ts [hunk 3]
```diff
@@ -11,14 +11,28 @@ import type { HostFolderFiledState } from './../host-folder-field.store';
...
             tap(() => patchState(store, { status: 'LOADING' })),
-            switchMap((path) => {
+            switchMap(({ path, isRequired }) => {
                 return dotEditContentService
                     .getSitesTreePath({ perPage: PEER_PAGE_LIMIT, filter: '*' })
                     .pipe(
+                        map((sites) => {
+                            if (isRequired) {
+                                return sites.filter((site) => site.label !== SYSTEM_HOST_NAME);
+                            }
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/methods/loadSites.ts:32]: Not sure about this... why required will remove SYSTEM_HOS?

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**